wire/sender.go

```diff
@@ -29,7 +29,7 @@ func SendMessageString(s Sender, msg string) error {
 }
 
 func (s *realSender) SendMessage(msg []byte) error {
-	if len(msg) > MaxMessageLength {
+	if len(msg) > MaxMessageLength - 4 {
```

Comment:
Why -4?

Reply:
A message's payload is combined from:

1. 4-byte hex string denotes message's length
2. The message itself

The ADB's [implementation](https://cs.android.com/android/platform/superproject/+/master:packages/modules/adb/adb_io.cpp;drc=90228a63bb6a59e8195165fbb7c332be27459696;l=39) implies that the `MaxPayloadSize` is the size of the message AND the length-string:

```
MaxPayloadSize = MessageLenghtStringSize + MaxMessageLength
```

Therefore we need to minus 4 bytes (length-string) to get the message's size.